Common/cpp/SharedItems/ShareableValue.cpp

```diff
@@ -72,62 +74,74 @@ void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType
   
   if (objectType == ValueType::MutableValueType) {
     type = ValueType::MutableValueType;
-    mutableValue = std::make_shared<MutableValue>(rt, value, module, module->scheduler);
+    valueContainer = std::make_unique<MutableValueWrapper>(
+      std::make_shared<MutableValue>(rt, value, module, module->scheduler)
+    );
   } else if (value.isUndefined()) {
     type = ValueType::UndefinedType;
   } else if (value.isNull()) {
     type = ValueType::NullType;
   } else if (value.isBool()) {
     type = ValueType::BoolType;
-    boolValue = value.getBool();
+    valueContainer = std::make_unique<BooleanValueWrapper>(value.getBool());
   } else if (value.isNumber()) {
     type = ValueType::NumberType;
-    numberValue = value.asNumber();
+    valueContainer = std::make_unique<NumberValueWrapper>(value.asNumber());
   } else if (value.isString()) {
     type = ValueType::StringType;
-    stringValue = value.asString(rt).utf8(rt);
+    valueContainer = std::make_unique<StringValueWrapper>(value.asString(rt).utf8(rt));
   } else if (value.isObject()) {
     auto object = value.asObject(rt);
     if (object.isFunction(rt)) {
       if (object.getProperty(rt, "__worklet").isUndefined()) {
         // not a worklet, we treat this as a host function
         type = ValueType::HostFunctionType;
-        hostRuntime = &rt;
-        hostFunction = std::make_shared<HostFunctionHandler>(std::make_shared<jsi::Function>(object.asFunction(rt)), rt);
         containsHostFunction = true;
+        valueContainer = std::make_unique<HostFunctionWrapper>(
+          std::make_shared<HostFunctionHandler>(std::make_shared<jsi::Function>(object.asFunction(rt)), rt),
+          &rt
+        );
       } else {
         // a worklet
         type = ValueType::WorkletFunctionType;
-        frozenObject = std::make_shared<FrozenObject>(rt, object, module);
-        containsHostFunction |= frozenObject->containsHostFunction;
+        std::shared_ptr<FrozenObject> frozenObject = std::make_shared<FrozenObject>(rt, object, module);
+        containsHostFunction = frozenObject->containsHostFunction;
```

Comment:
Why you discarded `or` operation here? I think it's safer to leave it that way(`|=` instead of `=`).